hyperSpec/R/palette_matlab.R

```diff
@@ -82,13 +82,39 @@ matlab.dark.palette <- function(n = 100, ...) {
 #' # Alois palette
 #'
 #' op <- par(mar = c(0, 0, 0, 0))
-#' pie(rep(1, 12), col = alois.palette(12))
+#' pie(rep(1, 12), col = palette_alois(12))
 #' par(op)
 #'
-#' plot(flu, col = alois.palette(nrow(flu)))
+#' plot(flu, col = palette_alois(nrow(flu)))
 #'
-#' plotmap(faux_cell[, , 1200], col.regions = alois.palette())
+#' plotmap(faux_cell[, , 1200], col.regions = palette_alois())
 #'
-alois.palette <- function(n = 100, ...) {
+palette_alois <- function(n = 100, ...) {
   colorRampPalette(c("black", "blue", "cyan", "green", "yellow", "red"), ...)(n)
 }
+
+# Unit tests -----------------------------------------------------------------
+#' @import hySpc.testthat
+hySpc.testthat::test(palette_matlab) <- function() {
+  context("palette_matlab")
+  test_that("palette_matlab() works", {
+    pal <- palette_matlab()
+    expect_true(pal[1] == "#0000FF" | pal[1] == "#0000FFFF")
+    expect_true(pal[2] == "#000AFF" | pal[2] == "#000AFFFF")})
+}
+
+hySpc.testthat::test(palette_matlab_dark) <- function() {
+  context("palette_matlab_dark")
+  test_that("palette_matlab_dark() generates correct colors", {
+    dark <- palette_matlab_dark()
+    expect_equal(dark[1], "#0000FF")
+    expect_equal(dark[2], "#0005FF")})
+}
+
+hySpc.testthat::test(palette_alois) <- function() {
+  context("palette_alois")
+  test_that("palette_alois() works", {
+    alois <- palette_alois()
+    expect_equal(alois[1], "#000000")
+    expect_equal(alois[2], "#00000C")})
+}
```

Comment:
Please, leave a blank line at the end of each file. In some circumstances, the absence of a new line may result in unwanted issues.

![image](https://user-images.githubusercontent.com/12725868/122726426-a7ed6f80-d27e-11eb-8665-5ecba5d13428.png)